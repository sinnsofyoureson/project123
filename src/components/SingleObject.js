import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Div = styled.div`
    display: flex;
    flex-direction: column;
    color: #fff;
    padding: 8px 34px;
    border-radius: 5px;
    cursor: pointer;
    transition: .2s ease-in-out;
    border-bottom: solid 1px #000;

    &:hover {
        background: #050505;
    }
    &:active {
        background: #a1a1a1;
        transition: .5s ease-in-out;

    }
`

const Paragraph = styled.p`
    color: #555;
    margin-right: 5px;
`

const List = styled.div`
    display: flex;
    margin: 0;
    padding: 0;
`

const Button = styled.button`
    cursor: pointer;
    margin: 3px 0 3px 13px;
    padding: 8px 21px;
    background: #080808;
    color: #e5e5e5;
    border: none;
    border-radius: 5px;
    font-size: 18px;
    font-weight: 100;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
    transition: .2s ease-in-out;


    &:hover {
        background: #e5e5e5;
        color: #080808;
        box-shadow: 0 0 8px #fff;
    }
    &:active {
        background: #000;
        color: #e5e5e5;
    }
`

const SingleObject = ({ showObjectInfo, onClick, id, type, address, client, contract, services, status, date, expiration, active, shown }) => (
    
    <li
        onClick={showObjectInfo}
        style={{ 
            display: active ? 'inline' : 'none' 
        }}
    >
        <Div>
            <List style={{ flexDirection: 'column' }}>
                <List>
                    <Paragraph>Договор</Paragraph>
                    <p>{contract}</p>
                </List>
                <List>
                    <Paragraph>тип: </Paragraph>
                    <p>{type}</p>
                </List>
                <List>
                    <Paragraph>от </Paragraph>
                    <p>{date}</p>
                </List>
            </List>
            <List style={{ display: shown ? 'flex' : 'none' }}>
                <Paragraph>адресс: </Paragraph><p>{address}</p>
            </List>
            <List style={{ display: shown ? 'flex' : 'none' }}>
                <Paragraph>клиент: </Paragraph><p>{client}</p>
            </List>
            <List style={{ display: shown ? 'flex' : 'none' }}>
                <Paragraph>подключенные услуги: </Paragraph><p>{services}</p>
            </List>
            <List style={{ display: shown ? 'flex' : 'none' }}>
                <Paragraph>статус:</Paragraph><p>{status}</p>
            </List>
            <List style={{ display: shown ? 'flex' : 'none' }}>
                <Paragraph>дата, до которой действует договор: </Paragraph><p>{expiration}</p>
            </List>
            <List style={{display: 'flex', justifyContent: 'flex-end' }}>
                <Button kety={id} style={{ display: shown ? 'flex' : 'none' }}>Update!</Button>
                <Button kety={id} onClick={onClick} style={{ display: shown ? 'flex' : 'none' }}>Delete!</Button>
            </List>
        </Div>

        
        
    </li>
)

SingleObject.propTypes = {
    showInfo: PropTypes.func,
    onClick: PropTypes.func,
    id: PropTypes.string,
    type: PropTypes.string,
    address: PropTypes.string,
    client: PropTypes.string,
    contract: PropTypes.string,
    services: PropTypes.string,
    status: PropTypes.string,
    date: PropTypes.string,
    expiration: PropTypes.string,
    active: PropTypes.bool,
    shown: PropTypes.bool
}

export default SingleObject
