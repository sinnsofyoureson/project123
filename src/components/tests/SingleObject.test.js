import React from 'react';
import renderer from 'react-test-renderer';

// components and containers testing
import SingleObject from '../SingleObject';

describe('SingleObject component', () => {
    it('matches the snapshot', () => {
      const props = {
        objects: []
      }
      const tree = renderer.create(<SingleObject {...props} />).toJSON();
      jestExpect(tree).toMatchSnapshot();
    })
});