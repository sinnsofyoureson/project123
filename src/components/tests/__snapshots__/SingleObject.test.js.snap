// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleObject component matches the snapshot 1`] = `
<li
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="sc-bdVaJa ggxMsz"
  >
    <div
      className="sc-htpNat kTHLBp"
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <div
        className="sc-htpNat kTHLBp"
      >
        <p
          className="sc-bwzfXH gtBSna"
        >
          Договор
        </p>
        <p />
      </div>
      <div
        className="sc-htpNat kTHLBp"
      >
        <p
          className="sc-bwzfXH gtBSna"
        >
          тип: 
        </p>
        <p />
      </div>
      <div
        className="sc-htpNat kTHLBp"
      >
        <p
          className="sc-bwzfXH gtBSna"
        >
          от 
        </p>
        <p />
      </div>
    </div>
    <div
      className="sc-htpNat kTHLBp"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <p
        className="sc-bwzfXH gtBSna"
      >
        адресс: 
      </p>
      <p />
    </div>
    <div
      className="sc-htpNat kTHLBp"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <p
        className="sc-bwzfXH gtBSna"
      >
        клиент: 
      </p>
      <p />
    </div>
    <div
      className="sc-htpNat kTHLBp"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <p
        className="sc-bwzfXH gtBSna"
      >
        подключенные услуги: 
      </p>
      <p />
    </div>
    <div
      className="sc-htpNat kTHLBp"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <p
        className="sc-bwzfXH gtBSna"
      >
        статус:
      </p>
      <p />
    </div>
    <div
      className="sc-htpNat kTHLBp"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <p
        className="sc-bwzfXH gtBSna"
      >
        дата, до которой действует договор: 
      </p>
      <p />
    </div>
    <div
      className="sc-htpNat kTHLBp"
      style={
        Object {
          "display": "flex",
          "justifyContent": "flex-end",
        }
      }
    >
      <button
        className="sc-bxivhb cOpcuA"
        style={
          Object {
            "display": "none",
          }
        }
      >
        Update!
      </button>
      <button
        className="sc-bxivhb cOpcuA"
        style={
          Object {
            "display": "none",
          }
        }
      >
        Delete!
      </button>
    </div>
  </div>
</li>
`;
